src/libs/history.c

```diff
@@ -825,6 +825,45 @@ static void _lib_history_create_style_button_clicked_callback(GtkWidget *widget,
   }
 }
 
+void gui_reset(dt_lib_module_t *self)
+{
+  const int32_t imgid = darktable.develop->image_storage.id;
+  if(!imgid) return;
+
+  gint res = GTK_RESPONSE_YES;
+
+  if(dt_conf_get_bool("ask_before_delete"))
+  {
+    const GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+
+    GtkWidget *dialog = gtk_message_dialog_new(
+        GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT, GTK_MESSAGE_QUESTION, GTK_BUTTONS_YES_NO,
+        _("do you really want to clear history of current image?"));
+#ifdef GDK_WINDOWING_QUARTZ
+    dt_osx_disallow_fullscreen(dialog);
+#endif
+
+    gtk_window_set_title(GTK_WINDOW(dialog), _("delete images' history?"));
```

Comment:
Just saw this now... Shouldn't this be :
`"delete image's history?"`

As we are speaking of a single image.

Reply:
darn. i missed that one. Not worth doing PR for such small change, so if you can please change to `"delete image's history"` :) I can do PR for couple other spelling mistakes that I did, so - whatever you prefer!

Reply:
I just submitted a PR to separate the prefs. I'll include it in that.

Reply:
#6016

Reply:
Fixed! Sorry didn't see your message @elstoc !

Reply:
And no worry, Git will handle that without trouble!

Reply:
That's ok it's in my PR as well so there aren't any conflicts